src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.NodeMapBuilder.cs

```diff
@@ -255,6 +255,28 @@ public override BoundNode VisitBinaryOperator(BoundBinaryOperator node)
                 throw ExceptionUtilities.Unreachable;
             }
 
+            public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
+            {
+                // For deconstruction declarations, the BoundLocals in the LeftVariables should not be added to the map
+                if (!node.IsDeclaration)
+                {
+                    VisitList(node.LeftVariables);
+                }
+
+                Visit(node.Right);
+                // don't map the deconstruction, conversion or assignment steps
+                return null;
+            }
+
+            public override BoundNode VisitForEachStatement(BoundForEachStatement node)
+            {
+                Visit(node.IterationVariableType);
+                Visit(node.Expression);
+                Visit(node.Body);
+                // don't map the deconstruction steps
+                return null;
+            }
```

Comment:
Is the override necessary? `DeconstructionOpt` is marked `SkipInVisitor="true"`.